### DISCUSSION TITLE
How can remove the row delete button if record match some conditions in ModelResource

### DISCUSSION THREAD

**Discussion by WillieOng-HK** [moonshine-software/moonshine#1427] [Q&A] [answered]
How can remove the row delete button if record match some conditions in Model Resource

**Comment by lee-to** [comment, marked as answer]
We already have a package based on spatie with support for version 3
https://github.com/SWEET1S/moonshine-roles-permissions


If you still want to implement it yourself, here are the materials

https://moonshine-laravel.com/en/docs/3.x/model-resource/authorization

https://moonshine-laravel.com/en/docs/3.x/security/authorization#additional_logic

You can also just hide the button
https://moonshine-laravel.com/en/docs/3.x/model-resource/buttons#delete

```php
protected function modifyDeleteButton(ActionButtonContract $button): ActionButtonContract
{
    return $button->canSee(fn($data) => true|false);
}
```
